### PR TITLE
Added initial recipe for vim

### PR DESCRIPTION
**Pull request by mariusvniekerk**

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
@conda-forge/help-c-cpp This is ready for review

File: recipes/vim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (chrisburr) [inline recipes/vim/meta.yaml]: These comments can be deleted

File: recipes/vim/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  url: https://github.com/vim/vim/archive/v{{ version }}.tar.gz
+  sha256: 6197407497c5cac832b4417ece6ed4f54c03336ae947130e2ad295e480f0fb5f
+
+build:
+  skip: true  # [win]
+  number: 0
+
+requirements:
+  build:
+    # if your project compiles code (such as a C extension) then add the required compilers as separate entries here.
```
Comment (chrisburr) [inline recipes/vim/meta.yaml]: Same

File: recipes/vim/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+test:
+  commands:
+    - vim --version
+
+about:
+  home: https://www.vim.org/
+  # vi(m) is licensed as charity-ware
+  license: OTHER
+  # license_file: LICENSE.txt
```
Comment (chrisburr) [inline recipes/vim/meta.yaml]: So far as I can tell the license isn't being packaged, I think you can find it here:
```suggestion
  license_file: runtime/doc/uganda.txt
```

**Comment by chrisburr** [comment]
Looks good to me, LGTM 👍

**Pull request merged**